pkg/azure/testhelp/providerspec.go

```diff
@@ -145,6 +145,12 @@ func (b *ProviderSpecBuilder) WithSecurityProfile(sec *api.AzureSecurityProfile)
 	return b
 }
 
+// WithCapacityReservation configures the capacity reservation settings for the VM
+func (b *ProviderSpecBuilder) WithCapacityReservation(capacityReservationConfig *api.CapacityReservation) *ProviderSpecBuilder {
```

Comment:
where is this function called?
Did you forget to add tests in `pkg/azure/provider/provider_test.go`?